multibranch/src/test/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinderTest.java

```diff
@@ -229,4 +231,56 @@
         });
     }
 
+    /**
+     * This test aims to verify that {@link WorkflowMultiBranchProject#CRITERIA} is being applied.
+     *
+     * @throws Exception
+     */
+    @Test public void criteriaInvoked() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                sampleGitRepo.init();
+                sampleGitRepo.write("Jenkinsfile", "node { echo 'Hello World' }");
+                sampleGitRepo.git("add", "Jenkinsfile");
+                sampleGitRepo.git("commit", "--all", "--message=flow");
+                WorkflowMultiBranchProject mp = story.j.jenkins.createProject(WorkflowMultiBranchProject.class, "p");
+                mp.getSourcesList().add(new BranchSource(new GitSCMSource(null, sampleGitRepo.toString(), "", "*", "", false), new DefaultBranchPropertyStrategy(new BranchProperty[0])));
+                WorkflowJob p = WorkflowMultiBranchProjectTest.scheduleAndFindBranchProject(mp, "master");
+                assertEquals(1, mp.getItems().size());
+                story.j.waitUntilNoActivity();
+                WorkflowRun b1 = p.getLastBuild();
+                assertEquals(1, b1.getNumber());
+                sampleGitRepo.git("rm", "Jenkinsfile");
+                sampleGitRepo.git("commit", "--all", "--message=remove");
+                WorkflowRun b = story.j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
+                story.j.assertLogContains("java.io.FileNotFoundException", b);
+            }
+        });
+    }
+
+    @Issue("JENKINS-30798")
+    @Test public void branchName() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                sampleGitRepo.init();
+                sampleGitRepo.write("Jenkinsfile", "node { echo 'Hello World 1' }");
+                sampleGitRepo.git("add", "Jenkinsfile");
+                sampleGitRepo.git("commit", "--all", "--message=flow");
+                sampleGitRepo.git("checkout", "-b", "dev/main");
```

Comment:
Branch names containing `/` were not _expected_ to work without a proper fix of [JENKINS-30744](https://issues.jenkins-ci.org/browse/JENKINS-30744), which you are not really testing here. Claiming this tests JENKINS-30798 is misleading because that exception has been observed without weird branch names. I would just delete this test for now.